src/utils/releases.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+
+const github = require('@actions/github')
+const { logInfo, logError } = require('../log')
+
+async function fetchLatestRelease(inputs) {
+  try {
+    logInfo('Fetching latest release')
+
+    const token = inputs['github-token']
+
+    const { owner, repo } = github.context.repo
+    const octokit = github.getOctokit(token)
+    const { data: latestRelease } = await octokit.rest.repos.getLatestRelease({
+      owner,
+      repo,
+    })
+
+    logInfo(
+      `Latest release fetched successfully with tag: ${latestRelease.tag_name}`
+    )
+
+    return latestRelease
+  } catch (err) {
+    if (err.message === 'Not Found') {
+      logInfo(`No previous releases found`)
+      return null
+    }
+
+    logError(err.message)
+    throw new Error(
+      `An error occurred while fetching the latest release: ${err.message}`
+    )
+  }
+}
+
+async function generateReleaseNotes(inputs, newVersion, latestVersion) {
```

Comment:
- no point in providing the whole inputs, this function only needs the token
- nit: let's not call it latestVersion, let's call it baseVersion